tests/solver-by-substitution.cc

```diff
@@ -702,22 +728,24 @@ BOOST_AUTO_TEST_CASE(hybrid_solver)
   solver.explicitConstraintSetHasChanged();
   BOOST_TEST_MESSAGE(solver << '\n');
 
-  // BOOST_CHECK_EQUAL(solver.solve<lineSearch::Backtracking  >(q), BySubstitution::SUCCESS);
+  BOOST_CHECK(solver.isSatisfied(q0));
 
-  Configuration_t tmp = qrand;
+  vector_t v (vector_t::Random(device->numberDof()));
+  v *= .1;
+  Configuration_t qrand(q0);
+  LiegroupElement g(qrand, device->configSpace());
+  g += v;
+  qrand = g.vector();
   BOOST_CHECK_EQUAL(solver.solve<Backtracking  >(qrand),
                     BySubstitution::SUCCESS);
-  qrand = tmp;
+  qrand = g.vector();
   BOOST_CHECK_EQUAL(solver.solve<ErrorNormBased>(qrand),
                     BySubstitution::SUCCESS);
-  qrand = tmp;
+  qrand = g.vector();
   BOOST_CHECK_EQUAL(solver.solve<FixedSequence>(qrand),
                     BySubstitution::SUCCESS);
-
-  vector_t dq (device->numberDof());
-  dq.setRandom();
-  qrand = tmp;
-  solver.projectVectorOnKernel (qrand, dq, tmp);
+  BOOST_CHECK_EQUAL(solver.solve<Constant>(qrand),
+                    BySubstitution::SUCCESS);
```

Comment:
Should we expect that the solver finds approximately `q0` ?

Reply:
Yes.